src/Frame.cpp

```diff
@@ -244,7 +244,7 @@ Frame *Frame::GetFrame(FrameId fId)
 {
 	PROFILE_SCOPED()
 
-	if (fId && fId.id() < s_frames.size())
+	if (fId && fId.id() >= 0 && static_cast<std::size_t>(fId.id()) < s_frames.size())
```

Comment:
I totally agree, I just avoided doing these kind of changes. I will change the return type of `fId.id()` and I will make all the related changes around.

Reply:
A question: `FrameId::m_id` now is `int`, and the invalid values are all negative numbers. Do you think that it could be better to use an `unsigned int` (or an `uint32_t` if you don't mind changing the size for particular platforms) and use the `numeric_limits::max()` as invalid value?

Reply:
Using `numeric_limits::max()` is an acceptable solution - when the class was originally written it used a negative number as invalid simply to preserve existing behavior, but the proposed behavior is more elegant functionality. And definitely use `uint32_t` - if we have more than 4 billion Frames we're looking at a memory consumption of potentially 1 TB...